File: HackerRank Solutions/Interview Preparation Kit/Sorting/Mark and Toys.java
import java.io.*;
import java.math.*;
import java.security.*;
import java.text.*;
import java.util.*;
import java.util.concurrent.*;
import java.util.regex.*;

public class Solution {

    // Complete the maximumToys function below.
    public static void main(String[] args) {
        Scanner input = new Scanner(System.in);
        int n = input.nextInt();
        int k = input.nextInt();
        int[] prices = new int[n];
        for(int i=0; i<n; i++){
            prices[i] = input.nextInt();
        }
        Arrays.sort(prices);
        int maxNoOfToys = 0;
        for(int i=0; i<n; i++){
            k -= prices[i];
            if(k > 0){
                maxNoOfToys++;
            }
            if(k <= 0){
                break;
            }
        }
        System.out.println(maxNoOfToys);
    }
}

File: HackerRank Solutions/Interview Preparation Kit/Sorting/Mark and Toys.cpp
#include <bits/stdc++.h>
using namespace std;

int main(){
    int n, k, i, maxNoOfToys=0;
    cin >> n >> k;
    int prices[n];
    for(int i=0; i<n; i++){
        cin >> prices[i];
    }
    sort(prices, prices+n);
    i=0;
    while(prices[i] < k){
        maxNoOfToys = maxNoOfToys + prices[i];
        if(maxNoOfToys > k){
            break;
        }
        i++;
    }
    cout << i << endl;
    return 0;
}
